### PR TITLE
drop support for <2 and replace `run_cmd` with `run_shell_cmd` in custom easyblock for scalapack (`scalapack.py`)

### PR DESCRIPTION
**Pull request by branfosj**
(created using `eb --new-pr`)

Oldest easyconfigs in the archive are for 2.x, so we should drop support for < 2.

### CODE REVIEW

**Comment by branfosj** [comment]
Test report by @branfosj
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _ScaLAPACK-2.2.0-gompi-2022b-fb.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0105u03a - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), Python 3.6.8
See https://gist.github.com/branfosj/05114b045ac0329a966eeb86f5067f1f for a full test report.

File: easybuild/easyblocks/s/scalapack.py [hunk 1]
```diff
@@ -229,7 +185,7 @@ def build_libscalapack_make(self):
         cmd = "%s make %s %s" % (self.cfg['prebuildopts'], paracmd, self.cfg['buildopts'])
 
         # Ignore exit code for parallel run
-        (out, _) = run_cmd(cmd, log_ok=False, log_all=False, simple=False)
+        run_cmd(cmd, fail_on_error=False)
```
Comment (boegel) [inline easybuild/easyblocks/s/scalapack.py:188]: @branfosj You haven't switched to `run_shell_cmd` here?

**Review by boegel — CHANGES_REQUESTED** [review]

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
@boegelbot please test @ jsc-zen3
EB_BRANCH="5.0.x"
EB_ARGS="ScaLAPACK-2.2.0-gompi-2023b-fb.eb --installpath /tmp/$USER/pr3106"

**Pull request merged**